Code/GraphMol/FMCS/testFMCS_Unit.cpp

```diff
@@ -528,6 +529,217 @@ void testTarget_no_10188_49064() {
   BOOST_LOG(rdInfoLog) << "\tdone" << std::endl;
 }
 
+ROMol *embed3DFromSmiles(const char* smiString, const ROMol *scaffoldNoHs,
+                         const int seed){
+  int cid;
+  std::map< int, RDGeom::Point3D > coordMap;
+  std::vector<MatchVectType> matches;
+
+  const Conformer& scaffoldConf = scaffoldNoHs->getConformer(0);
+  ROMol *mol = SmilesToMol(smiString);
+  mol = MolOps::addHs(*mol);
+  int numMatches = RDKit::SubstructMatch(*mol, *scaffoldNoHs, matches);
+  TEST_ASSERT(numMatches > 0);
+
+  for (std::vector<MatchVectType>::const_iterator it = matches.begin();
+       it != matches.end(); ++it) {
+    MatchVectType matchVect = *it;
+    for (unsigned i = 0; i < matchVect.size(); i++){
+      unsigned scaffoldIdx = matchVect[i].first;
+      unsigned molIdx = matchVect[i].second;
+      coordMap[molIdx] = scaffoldConf.getAtomPos(scaffoldIdx);
+    }
+  }
+  RDKit::DGeomHelpers::EmbedParameters params(RDKit::DGeomHelpers::ETKDG);
+  params.coordMap = &coordMap;
+  params.useRandomCoords = true;
+  params.maxIterations = 10;
+  params.randomSeed = seed;
+  params.verbose = true;
+  cid = RDKit::DGeomHelpers::EmbedMolecule(*mol, params);
+  if (cid <= -1){
+    std::cerr << "cid was " << cid << " trying to embed: " <<std::endl;
+    for (std::map<int, RDGeom::Point3D>::const_iterator i=coordMap.begin();
+         i!=coordMap.end();i++){
+      std::cerr << i->first<<": "<<i->second<<std::endl;
+    }
+  }
+  TEST_ASSERT(cid > -1);
+  return mol;
+}
+#define MCSTESTREPEATS 0 // To run MCS repeatedly to measure performance
+MCSResult checkMCS(const std::vector<ROMOL_SPTR> mols, const MCSParameters p,
+              unsigned expectedAtoms, unsigned expectedBonds){
+  t0 = nanoClock();
+  MCSResult res = findMCS(mols, &p);
+  //std::shared_ptr<RWMol>
+  ROMol *mcsMol = SmartsToMol(res.SmartsString);
+  ROMol *cleanMCSMol = SmilesToMol(MolToSmiles(*mcsMol));
+  std::cout << "MCS: " << res.SmartsString << " " << MolToSmiles(*cleanMCSMol)
+            << " " << res.NumAtoms << " atoms, " << res.NumBonds << " bonds"
+            << std::endl;
+#ifdef MCSTESTREPEATS
+  for (int i=0; i<MCSTESTREPEATS; i++){
+    res = findMCS(mols, &p);
+  }
+#endif
+  printTime();
+  if (res.NumAtoms != expectedAtoms || res.NumBonds != expectedBonds){
+    std::cerr << "testMaxDistance failed, expected "
+              << expectedAtoms << " atoms, "<< expectedBonds << " bonds"
+              << std::endl;
+    TEST_ASSERT(res.NumAtoms == expectedAtoms && res.NumBonds == expectedBonds);
+  }
+  return res;
+}
+
+ROMol *scaffoldFromSmiles(const char *scaffoldSmiles, const int seed){
+  int cid;
+  const ROMol *scaffoldOrig = SmilesToMol(scaffoldSmiles);
+  ROMol *scaffold = MolOps::addHs(*scaffoldOrig);
+  cid = RDKit::DGeomHelpers::EmbedMolecule(*scaffold, 0, seed);
+  TEST_ASSERT(cid > -1);
+  return MolOps::removeHs(*scaffold);
+}
+/* TODO: best practice on where to put a test data file into the repo? */
```

Comment:
I'm not sure if I should keep this test, or if so, if there's a good place to put the input data file